crates/shared/src/order_validation.rs

```diff
@@ -892,14 +892,33 @@ fn minimum_balance(order: &OrderData) -> Option<U256> {
 /// Retrieves the quote for an order that is being created and verify that its
 /// fee is sufficient.
 ///
+/// The fee is checked only if `fee_amount` is specified.
+pub async fn get_quote_and_check_fee(
+    quoter: &dyn OrderQuoting,
+    quote_search_parameters: &QuoteSearchParameters,
+    quote_id: Option<i64>,
+    fee_amount: Option<U256>,
+) -> Result<Quote, ValidationError> {
+    let quote = get_quote(quoter, quote_search_parameters, quote_id).await?;
+
+    if let Some(fee_amount) = fee_amount {
+        if fee_amount < quote.fee_amount {
+            return Err(ValidationError::InsufficientFee);
+        }
+    }
```

Comment:
`Option`s have a bunch of nice little helper functions for these things.
Just wanted to point it out. Feel free to ignore if you prefer the pattern matching version.

```suggestion
    if fee_amount.is_some_and(|fee| fee < quote.fee_amount) {
        return Err(ValidationError::InsufficientFee);
    }
```

Reply:
Tnx, applied.